api/nodes/permissions.py

```diff
@@ -74,6 +75,32 @@ def has_object_permission(self, request, view, obj):
             has_auth = parent_node.can_edit(auth)
             return has_auth
 
+class ContributorOrPublicForRelationshipPointers(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, dict)
+        auth = get_user_auth(request)
+        parent_node = obj['self']
+
+        if request.method in permissions.SAFE_METHODS:
+            return parent_node.can_view(auth)
+        elif request.method == 'DELETE':
+            return parent_node.can_edit(auth)
+        else:
+            pointer_nodes = []
+            for pointer in request.data.get('data', []):
+                node = Node.load(pointer['id'])
+                if not node or node.is_folder:
+                    raise exceptions.NotFound
+                pointer_nodes.append(node)
+            has_parent_auth = parent_node.can_edit(auth)
```

Comment:
You should be able to bail out early (i.e. return `False`) if this is `False`.

But again, is this supposed to check admin permissions rather than r+w only?